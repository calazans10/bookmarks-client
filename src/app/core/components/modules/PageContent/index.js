import React from 'react';
import PropTypes from 'prop-types';
import Logo from '../Logo';
import MainWrapper from '../MainWrapper';
import MainHeader from '../MainHeader';
import MainNavigation from '../MainNavigation';
import MainContent from '../MainContent';

const PageContent = ({ to, children }) => (
  <MainWrapper>
    <MainHeader>
      <Logo to={to} />
      <MainNavigation />
    </MainHeader>
    <MainContent>{children}</MainContent>
  </MainWrapper>
);

PageContent.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default PageContent;
