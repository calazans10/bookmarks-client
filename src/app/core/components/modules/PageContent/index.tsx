import React from 'react';
import Logo from '../Logo';
import MainWrapper from '../MainWrapper';
import MainHeader from '../MainHeader';
import MainNavigation from '../MainNavigation';
import MainContent from '../MainContent';

type PageContentProps = {
  to: string;
  children: React.ReactNode;
};

const PageContent = ({ to, children }: PageContentProps) => (
  <MainWrapper>
    <MainHeader>
      <Logo to={to} />
      <MainNavigation />
    </MainHeader>
    <MainContent>{children}</MainContent>
  </MainWrapper>
);

export default PageContent;
