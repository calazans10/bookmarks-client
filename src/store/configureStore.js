import { applyMiddleware, createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { routerMiddleware } from 'connected-react-router';
import storage from 'redux-persist/lib/storage';
import logger from 'redux-logger';
import history from '../history';
import rootReducer from '../reducers';
import rootSaga from '../sagas';

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['ui'],
};

const createAppMiddleware = sagaMiddleware => {
  if (process.env.REACT_APP_ENVIRONMENT === 'development') {
    return composeWithDevTools(applyMiddleware(routerMiddleware(history), sagaMiddleware, logger));
  }

  return applyMiddleware(routerMiddleware(history), sagaMiddleware);
};

const sagaMiddleware = createSagaMiddleware();
const appMiddleware = createAppMiddleware(sagaMiddleware);

const persistedReducer = persistReducer(persistConfig, rootReducer);

export default () => {
  const store = createStore(persistedReducer, appMiddleware);
  const persistor = persistStore(store);
  sagaMiddleware.run(rootSaga);
  return { store, persistor };
};
