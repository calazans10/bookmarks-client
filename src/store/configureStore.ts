import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { persistStore, persistReducer } from 'redux-persist';
import createSagaMiddleware from 'redux-saga';
import { routerMiddleware } from 'connected-react-router';
import storage from 'redux-persist/lib/storage';
import logger from 'redux-logger';
import history from '../history';
import rootReducer from '../reducers';
import rootSaga from '../sagas';

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['ui'],
};

const createAppMiddleware = sagaMiddleware => {
  const appMiddleware = [
    ...getDefaultMiddleware({
      thunk: false,
      serializableCheck: false,
    }),
    routerMiddleware(history),
    sagaMiddleware,
  ];

  if (process.env.REACT_APP_ENVIRONMENT === 'development') {
    return [...appMiddleware, logger];
  }

  return appMiddleware;
};

const sagaMiddleware = createSagaMiddleware();
const middleware = createAppMiddleware(sagaMiddleware);

const persistedReducer = persistReducer(persistConfig, rootReducer);

export default () => {
  const store = configureStore({
    reducer: persistedReducer,
    middleware,
    devTools: process.env.REACT_APP_ENVIRONMENT !== 'production',
  });
  const persistor = persistStore(store);
  sagaMiddleware.run(rootSaga);
  return { store, persistor };
};
