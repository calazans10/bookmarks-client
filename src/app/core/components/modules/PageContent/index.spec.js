import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import PageContent from './index';

const mockStore = configureStore([]);

describe('PageContent', () => {
  it('renders without crashing', () => {
    const store = mockStore({
      ui: { alert: { message: '', isVisible: false } },
    });

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <PageContent to="/home">
            <p>Lorem Ipsum</p>
          </PageContent>
        </MemoryRouter>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
