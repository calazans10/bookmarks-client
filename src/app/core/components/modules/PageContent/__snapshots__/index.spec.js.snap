// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent renders without crashing 1`] = `
<div
  class="sc-AykKE cbnKn"
>
  <div
    class="alert"
  >
    <p
      class="alert__message"
    >
      <span
        class="alert__icon"
      />
      
    </p>
    <button
      aria-label="Close alert"
      class="alert__trigger"
      type="button"
    >
      <span
        class="sr-only"
      >
        x
      </span>
    </button>
  </div>
  <header
    class="sc-AykKF gPfWA-d"
  >
    <div
      class="sc-AykKG gQpYtq"
    >
      <div
        class="sc-AykKC gagnNP"
      >
        <a
          class="sc-AykKD inQTEp"
          href="/home"
        >
          Bookmarks
        </a>
      </div>
      <nav
        class="sc-AykKH cNaBMr"
      >
        <ul>
          <li
            class="sc-AykKI dyYYSk"
          >
            <button
              class="sc-AykKJ dkmSlc"
              type="button"
            >
              Sign Out
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <main
    class="sc-AykKK eIQyZi"
  >
    <p>
      Lorem Ipsum
    </p>
  </main>
</div>
`;
